### PR TITLE
CKKSVector: update replicate_first_slot

### LINKED ISSUE #225: Support replicate_first_slot for chunked vectors
## Feature Description
Right, for input that doesn't fit the slot_count, we split in chunks that fit it and encrypt each one to a ciphertext.
This flow breaks some operations like matmul_plain.

replicate_first_slot is also disabled right now, and we should enable it for CKKSVector and BFVVector.

### PR DESCRIPTION
**Pull request by youben11**
## Description
Solved for CKKSVector only

Fixes #225 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tenseal/cpp/tensors/ckksvector.cpp [hunk 1]
```diff
@@ -510,27 +509,52 @@ shared_ptr<CKKSVector> CKKSVector::enc_matmul_plain_inplace(
...
+    };
+
+    // recreate ciphertexts
+    this->_ciphertexts = vector<Ciphertext>();
+    vector<size_t> sizes = vector<size_t>();
+    // number of full replicated ciphertexts
+    auto n_full = n / slot_count;
+    // number of slots used for the last ciphertext
+    auto n_remain = n % slot_count;
+    this->_ciphertexts.reserve(n_full + n_remain == 0 ? 0 : 1);
```
Comment (bcebere) [inline tenseal/cpp/tensors/ckksvector.cpp:540]: I hope nobody evaluates `n_full + n_remain` first.

File: tenseal/cpp/tensors/ckksvector.cpp [hunk 2]
```diff
@@ -510,27 +509,52 @@ shared_ptr<CKKSVector> CKKSVector::enc_matmul_plain_inplace(
...
-    Ciphertext tmp = this->_ciphertexts[0];
     auto galois_keys = this->tenseal_context()->galois_keys();
-    for (size_t i = 0; i < (size_t)ceil(log2(n)); i++) {
-        this->tenseal_context()->evaluator->rotate_vector_inplace(
-            tmp, static_cast<int>(-pow(2, i)), *galois_keys);
-        this->tenseal_context()->evaluator->add_inplace(this->_ciphertexts[0],
-                                                        tmp);
-        tmp = this->_ciphertexts[0];
+
+    auto replicator = [&](Ciphertext ct, size_t n_repl) -> Ciphertext {
```
Comment (bcebere) [inline tenseal/cpp/tensors/ckksvector.cpp:521]: do we need a copy of `ct` here? instead of ref.

**Review by bcebere — APPROVED** [review]
LGTM!

File: tenseal/cpp/tensors/ckksvector.cpp (hunk 1 above)
Comment (youben11) [reply tenseal/cpp/tensors/ckksvector.cpp:540]: Will use parentheses

File: tenseal/cpp/tensors/ckksvector.cpp (hunk 2 above)
Comment (youben11) [reply tenseal/cpp/tensors/ckksvector.cpp:521]: Yeah, we will be creating an additional ciphertext if we didn't.

**Pull request merged**